cli/actions/synthesize_solvable_action.py

```diff
@@ -0,0 +1,62 @@
+from argparse import Namespace
+from .action import Action
+from cli.common import prepare_program, parse_program
+from termcolor import colored
+from program.solvable_synthesizer import SolvableSynthesizer
+from program import Program
+from program.assignment import PolyAssignment
+
+
+class SynthSolvAction(Action):
+    cli_args: Namespace
+
+    def __init__(self, cli_args: Namespace):
+        self.cli_args = cli_args
+
+    def __call__(self, *args, **kwargs):
+        benchmark = args[0]
+        combination_deg = self.cli_args.synth_solv
+        program = parse_program(benchmark, self.cli_args.transform_categoricals)
+        program = prepare_program(program, self.cli_args)
+
+        if len(program.non_mc_variables) == 0:
+            print(f"Loop in {benchmark} is already solvable.")
+            return
+
+        combination_vars = []
+        for var in program.non_mc_variables:
+            if var in program.original_variables:
+                combination_vars.append(var)
+
+        print(colored("-------------------", "cyan"))
+        print(colored("- Analysis Result -", "cyan"))
+        print(colored("-------------------", "cyan"))
+        print()
+
+        solvable_programs = []
+        combinations = SolvableSynthesizer.synthesize(combination_vars, combination_deg, program)
+        if combinations is None:
+            print(f"No combination found with degree {combination_deg}. Try using higher degrees.")
+            return program
+        else:
+            for combination in combinations:
+                solvable_loop_body = []
```

Comment:
Could you move the construction of the program to the SolvableSynthesizer? The CLI actions should do as minimal logic as possible. Doing so we can use the synthesizer in other contexts in the future (without CLI)